src/common/config/decrypt.tsx

```diff
@@ -48,25 +50,42 @@ export const getGsnRelaySigner = async (
 
   // Decrypt wallet to use wallet account as signer
   gsnProvider.addAccount(account.privateKey);
-  const from = account.address;
+  const from = await account.getAddress();
 
   // GsnProvider is now an rpc provider with GSN support. make it an ethers provider:
   const etherProvider = new providers.Web3Provider(gsnProvider);
 
   return etherProvider.getSigner(from);
 };
 
-export const decryptWallet = async (
+export const decryptWalletOrSigner = async (
   config: ConfigFile,
   password: string,
   progressCallback: (progress: number) => void
-): Promise<Wallet> => {
-  const decryptedWallet = await Wallet.fromEncryptedJson(config.wallet, password, progressCallback);
-  if (config.network === "local") {
-    const provider = new providers.JsonRpcProvider();
-    return decryptedWallet.connect(provider);
-  } else {
-    const connectedWallet = await decryptedWallet.connect(getDefaultProvider(config.network));
+): Promise<Wallet | ConnectedSigner> => {
+  const provider = config.network === "local" ? new providers.JsonRpcProvider() : getDefaultProvider(config.network);
+
+  if (isWalletOption(config.wallet)) {
+    const decryptedWallet = await Wallet.fromEncryptedJson(config.wallet, password, progressCallback);
+    const connectedWallet = await decryptedWallet.connect(provider);
     return connectedWallet;
+  } else {
+    const kmsCredentials = {
+      accessKeyId: config.wallet.accessKeyId, // credentials for your IAM user with KMS access
+      secretAccessKey: password, // credentials for your IAM user with KMS access
+      region: config.wallet.region,
+      keyId: config.wallet.kmsKeyId,
+    };
+
+    const signer = new AwsKmsSigner(kmsCredentials).connect(provider);
+    try {
+      const connectedSigner = signer as ConnectedSigner;
+      if ((await connectedSigner.getTransactionCount()) > -1) {
+        return connectedSigner;
```

Comment:
is this logic right? what does it mean if a `connectedSigner` has negative transactions?

Reply:
I am actually using the getTransactionCount to check if the wallet decrypt is successful. If the wallet is correct it will return TransactionCount which may span from 0 to x amount of transaction. If the wallet is incorrect, the getTransactionCount will throw an error

Reply:
thus as long as the wallet is able to get transaction count, I will assume that the wallet is connected correctly :x

Reply:
https://github.com/Open-Attestation/open-attestation-cli/blob/d765d15987dbb08a772a15f6befd915987484b5a/src/implementations/utils/wallet.ts#L72

the cli seems to just check whether a provider is available, would that be sufficient?

Reply:
that isn't sufficient, both valid and invalid passwords will have provider properties. if u test the oa-cli, it only returns the error of invalid wallet at deploy function (if u test the doc store deploy) instead of the entry password function. So far I don't see any situation where provider doesn't exist, but I place the logic check there just in case there is an undiscovered path.

Reply:
Changed the validation from getTransactionCount to getAddress instead